src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyController.java

```diff
@@ -174,19 +175,28 @@ public Study finishStudyLogo(@PathVariable String id, @PathVariable("createdOn")
     }
 
     @GetMapping(path = "/v1/apps/{appId}/studies/{studyId}", produces = { APPLICATION_JSON_UTF8_VALUE })
-    public String getStudyForApp(@PathVariable String appId, @PathVariable String studyId)
+    public String getStudyForApp(@PathVariable String appId, @PathVariable String studyId,
+            @RequestParam(required = false) String full)
             throws JsonProcessingException {
-        CacheKey key = CacheKey.publicStudy(appId, studyId);
-        String json = cacheProvider.getObject(key, String.class);
-        if (json == null) {
+        boolean fullBool = Boolean.parseBoolean(full);
+        if (fullBool) {
+            getAuthenticatedSession(WORKER);
             appService.getApp(appId);
```

Comment:
Add comment: This will throw if the app doesn't exist, with an error that says the app doesn't exist.